pallets/rolldown/src/tests.rs

```diff
@@ -350,3 +298,48 @@ fn test_conversion_address() {
 
 	assert_eq!(DummyAddressConverter::convert(byte_address), consts::CHARLIE);
 }
+
+#[test]
+#[serial]
+fn withdraw() {
+	ExtBuilder::new()
+		.issue(ALICE, ETH_TOKEN_ADDRESS_MGX, MILLION)
+		.execute_with_default_mocks(|| {
+			Rolldown::withdraw(
+				RuntimeOrigin::signed(ALICE),
+				ETH_RECIPIENT_ACCOUNT,
+				ETH_TOKEN_ADDRESS,
+				1_000_000u128,
+			)
+			.unwrap();
+
+			let withdraw_update = Withdraw {
+				withdrawRecipient: ETH_RECIPIENT_ACCOUNT,
+				tokenAddress: ETH_TOKEN_ADDRESS,
+				amount: U256::from(1_000_000u128),
+			};
+			assert_eq!(TokensOf::<Test>::free_balance(ETH_TOKEN_ADDRESS_MGX, &ALICE), 0_u128);
+			assert_eq!(
+				pending_updates::<Test>::get(sp_core::U256::from(u128::MAX / 2 + 1)),
+				Some(PendingUpdate::Withdraw(withdraw_update))
+			);
+		});
+}
+
+#[test]
+#[serial]
+fn error_on_withdraw_more() {
+	ExtBuilder::new()
+		.issue(ALICE, ETH_TOKEN_ADDRESS_MGX, MILLION)
+		.execute_with_default_mocks(|| {
+			assert_err!(
+				Rolldown::withdraw(
+					RuntimeOrigin::signed(ALICE),
+					ETH_RECIPIENT_ACCOUNT,
+					ETH_TOKEN_ADDRESS,
+					10_000_000u128
+				),
+				Error::<Test>::NotEnoughAssets
+			);
+		});
+}
```

Comment:
please add ut that makes sure that now with new request_id the tx will be removed from the list of pending updates after proper l1update is received

Reply:
i think it will be rejected because of too high request_id or sth